archive/tests.py

```diff
@@ -4,44 +4,58 @@
 from django.core.files.uploadedfile import SimpleUploadedFile
 from django.test import TestCase
 from django.utils import timezone
+from PIL import Image
 
 from archive.models import TYPE_CHOICES, Collection, Document, Picture
 
 
-# models tests
-class CollectionTestCase(TestCase):
+def create_collection(title="Test collection", collection_type=None):
+    return Collection.objects.create(title=title, pub_date=timezone.now(), type=collection_type)
+
+
+def create_picture(favorite=False):
+    collection = create_collection(collection_type=TYPE_CHOICES[0][1])
+    img = Image.new('RGB', (100, 100))
+    img.save(os.path.join(settings.MEDIA_ROOT, 'test_image.jpg'))
+    img.close()
+    img_file = open(os.path.join(settings.MEDIA_ROOT, 'test_image.jpg'),"rb")
+    img_data = img_file.read()
+    img_file.close()
+    test_image = SimpleUploadedFile(name='test_image.jpg',
+                                    content=img_data,
+                                    content_type='image/jpg')
+    return Picture.objects.create(collection=collection, image=test_image, favorite=favorite)
+
 
-    def create_collection(self, title="Test collection", collection_type=None):
-        return Collection.objects.create(title=title, pub_date=timezone.now(), type=collection_type)
+def create_document(title="Test document"):
+    collection = create_collection(collection_type=TYPE_CHOICES[1][1])
+    doc_file = open(os.path.join(settings.MEDIA_ROOT, 'test_document.doc'), "rb")
```

Comment:
This tests seems to fail if we do not have the file.

Chat GPT gave me this nice code of using temp files which seems to work:

```
import tempfile


def create_document(title="Test document"):
    collection = create_collection(collection_type=TYPE_CHOICES[1][1])

    # Create a temporary file with some test data
    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
        temp_file.write(b'This is some test data for the document')
        temp_file.seek(0)  # Reset the file pointer to the beginning
        test_document = SimpleUploadedFile(name='test_document.doc', content=temp_file.read())

    return Document.objects.create(collection=collection, title=title, document=test_document)
```

Reply:
True, that's a nice solution since fixing this with just the "open" function is unnecessarily messy. I swapped out the code and it seems to be working now.